data-prepper-plugins/common/src/main/java/org/opensearch/dataprepper/plugins/sink/FileSink.java

```diff
@@ -117,7 +121,13 @@ public void shutdown() {
     @Override
     public void initialize() {
         try {
-            writer = Files.newBufferedWriter(Paths.get(outputFilePath), StandardCharsets.UTF_8);
+            if (appendMode) {
+
+
+                writer = Files.newBufferedWriter(Paths.get(outputFilePath), StandardCharsets.UTF_8, StandardOpenOption.APPEND, StandardOpenOption.CREATE, StandardOpenOption.WRITE);
```

Comment:
I think we should try to consolidate this call a little bit. This way, we are sure we use the same parameters between these calls.

```
final OpenOption[] openOptions = appendMode ? 
    new OpenOption[] {StandardOpenOption.APPEND, StandardOpenOption.CREATE, StandardOpenOption.WRITE} : 
    new OpenOption[] {};
writer = Files.newBufferedWriter(Paths.get(outputFilePath), StandardCharsets.UTF_8, openOptions);
```